projects/ng2-smart-table/src/lib/ng2-smart-table.component.ts

```diff
@@ -108,6 +113,38 @@ export class Ng2SmartTableComponent implements OnChanges {
     this.rowClassFunction = this.grid.getSetting('rowClassFunction');
   }
 
+  selectRow(index: number, switchPageToSelectedRowPage: boolean = this.grid.getSetting('switchPageToSelectedRowPage')): void {
+    if (!this.grid) {
+      return;
+    }
+    const dataAmount: number = this.source?.data?.length;
+    if (index < 0 || (typeof dataAmount === 'number' && index >= dataAmount)) {
+      this.grid.dataSet.deselectAll();
+      return;
+    }
+
+    if (switchPageToSelectedRowPage) {
+      const source: DataSource = this.source;
+      const paging: { page: number, perPage: number } = source.getPaging();
+      const page: number = Math.ceil((index + 1) / paging.perPage);
+
+      if (page !== paging.page) {
+        (this.grid.settings as any).selectedRowIndex = index;
```

Comment:
Oh, I think now I get why `getRowIndexToSelect` could return `selectedRowIndex % perPage`. Am I correct? If so, let's pass it in some other way as it's too hard to grasp. Probably, we could add some internal used field with an index to set after page change?

Reply:
in this line, we change the value of the selectedRowIndex directly so that after due to the call of the SetPage method, in calculating the desired index for allocation, this one is used, because when calling the selectRow method in the table, we do not change the settings of the table itself.

Reply:
but the reason for returning % perPage I described this https://github.com/akveo/ng2-smart-table/pull/1210/files#r499736901